### DISCUSSION TITLE
Support multiple output target paths from BSP

### DISCUSSION THREAD

**Discussion by aishfenton** [build-server-protocol/build-server-protocol#498] [Proposals]
For build tools that support large monorepos (such as Bazel, Blaze, Pants, Buck, etc) it's typical to have multiple output paths, per build target. Compilation results are partitioned into different chunks (for different subdirectories), and stored in different places across the filesystem.

Currently the specification either implies that a single output path is used per BuildTarget such as in [ScalacOptionsItem.classDirectory](https://build-server-protocol.github.io/docs/extensions/scala#scalac-options-request), or it isn't specified.

The new addition of [buildTarget/outputPaths](https://build-server-protocol.github.io/docs/specification#output-paths-request) sounds like it is meant to support this. But on reading the original [issue](https://github.com/build-server-protocol/build-server-protocol/pull/269), it seems it was meant only to notify the IDE to exclude certain paths from indexing.

**Comment by aishfenton** [comment]
Related to https://github.com/scalameta/metals-feature-requests/issues/331

**Comment by lefou** [comment]
I strongly support this request.

In Mill, I also found the single `URI` in the `classDirectory` parameter (Java, Scala) quite limiting. Mill is producing the semanticDB data files into a different directory that the `*.class` files and this single `URI` requirement forces us to copy all files into a third directory just for consumption by BSP.

**Comment by lefou** [comment]
It would be nice to get some feedback from any who is involved in BSP development and maintenance. 

I'd love to see support for multiple classes directories since this would avoid unnecessary copy operations for each Java and Scala module (at least for Metals, which requires semanticDB). And since this also affects Bazel users in general, I'd expect consensus or some discussion.
